components/vercel/actions/create-deployment/create-deployment.mjs

```diff
@@ -0,0 +1,63 @@
+import vercel from "../../vercel.app.mjs";
+
+export default {
+  key: "vercel-create-deployment",
+  name: "Create Deployment",
+  description: "Create a new deployment with all the required and intended data. [See the docs](https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    vercel,
+    name: {
+      type: "string",
+      label: "Name",
+      description: "A string with the project name used in the deployment URL",
+    },
+    project: {
+      propDefinition: [
+        vercel,
+        "project",
+      ],
+      description: "The target project identifier in which the deployment will be created. When defined, this parameter overrides name",
+    },
+    repoId: {
+      type: "string",
+      label: "Git Source Repository Id",
+      description: "Id of the source repository",
+    },
+    branch: {
+      type: "string",
+      label: "Branch",
+      description: "Branch of repository to deploy to",
+      default: "master",
+    },
```

Comment:
I agree it would be great to have async options for the `repoId` and `branch`. I looked for a way to retrieve them from Vercel's API, but didn't find a way. I think it could get a little messy importing components between apps, and as you mentioned, user's can use the GitHub action and pass the reference along.